kong/runloop/handler.lua

```diff
@@ -1149,6 +1150,53 @@ return {
       STREAM_TLS_TERMINATE_SOCK = fmt("unix:%s/stream_tls_terminate.sock", prefix)
       STREAM_TLS_PASSTHROUGH_SOCK = fmt("unix:%s/stream_tls_passthrough.sock", prefix)
 
+      if is_http_module then
+        -- if worker has outdated log level (e.g. newly spawned), updated it
+        timer_at(0, function()
+          local cur_log_level = get_sys_filter_level()
+          local shm_log_level = ngx.shared.kong_log_level:get("level")
+          if cur_log_level and shm_log_level and cur_log_level ~= shm_log_level then
+            local ok, err = pcall(set_log_level, shm_log_level)
+            if not ok then
+              log(ERR, "failed setting log level for new worker: ", err)
+            end
+          end
+        end)
+
+        -- log level cluster event updates
+        kong.cluster_events:subscribe("log_level", function(data)
+          log(NOTICE, "log level cluster event received")
+
+          if not data then
+            kong.log.err("received empty data in cluster_events subscription")
+            return
+          end
+
+          local ok, err = kong.worker_events.post("debug", "log_level", tonumber(data))
+
+          if not ok then
+            kong.log.err("failed broadcasting to workers: ", err)
+            return
+          end
+
+          log(NOTICE, "log level event posted for node")
+        end)
+
+        -- log level worker event updates
+        kong.worker_events.register(function(data)
+          log(NOTICE, "log level worker event received")
+
+          local ok, err = pcall(set_log_level, data)
+
+          if not ok then
+            log(ERR, "[events] could not broadcast log level event: ", err)
```

Comment:
This error message is inaccurate - the error refers to the `set_log_level` call.